storage/mysql/log_storage_test.go

```diff
@@ -1092,32 +972,46 @@ func createTestLeaves(n, startSeq int64) []trillian.LogLeaf {
 }
 
 // Convenience methods to avoid copying out "if err != nil { blah }" all over the place
-func commit(tx storage.LogTreeTX, t *testing.T) {
-	if err := tx.Commit(); err != nil {
-		t.Fatalf("Failed to commit tx: %v", err)
+func beginLogTx(s storage.LogStorage, logID int64, t *testing.T) storage.LogTreeTX {
+	tx, err := s.BeginForTree(context.Background(), logID)
+	if err != nil {
+		t.Fatalf("Failed to begin log tx: %v", err)
 	}
+	return tx
 }
 
-func beginLogTx(s storage.LogStorage, logID logIDAndTest, t *testing.T) storage.LogTreeTX {
-	tx, err := s.BeginForTree(context.Background(), logID.logID)
+type committableTX interface {
+	Commit() error
+}
 
-	if err != nil {
-		t.Fatalf("Failed to begin log tx: %v", err)
+func commit(tx committableTX, t *testing.T) {
+	if err := tx.Commit(); err != nil {
+		t.Errorf("Failed to commit tx: %v", err)
 	}
+}
 
-	return tx
+type closeableTX interface {
+	Rollback() error
+	IsOpen() bool
 }
 
-func failIfTXStillOpen(t *testing.T, op string, tx storage.LogTreeTX) {
-	if r := recover(); r != nil {
-		// Check for the test bailing with panic before testing for unclosed tx.
-		// debug.Stack() does the right thing and includes the original failure point
-		t.Fatalf("Panic in %s: %v %v", op, r, string(debug.Stack()))
+// TODO(codingllama): Consider providing Close() methods to all TX types.
+func closeTX(tx closeableTX) error {
```

Comment:
Right, but wouldn't the old code have generated test failures if various places if the Tx was still open by end of the test, but that now wouldn't get caught?  Or have I misunderstood what `failIfTXStillOpen` was doing?